app/services/order_details/reconciler.rb

```diff
@@ -58,29 +58,26 @@ def reconcile_all
     def unreconcile_all
       @count = 0
       @persist_errors = []
-      rollback_occurred = false
-
-      OrderDetail.transaction do
-        order_details.each do |order_detail|
-          next unless order_detail.reconciled?
 
-          begin
-            order_detail.update!(
-              state: "complete",
-              order_status: OrderStatus.complete,
-              reconciled_at: nil,
-              deposit_number: nil
-            )
-            @count += 1
-          rescue => e
-            @persist_errors << "Order ##{order_detail.id}: #{e.message}"
-            rollback_occurred = true
-            raise ActiveRecord::Rollback
-          end
+      order_details.each do |order_detail|
+        next unless order_detail.reconciled?
+
+        begin
+          order_detail.update_columns(
+            state: "complete",
+            order_status_id: OrderStatus.complete.id,
+            reconciled_at: nil,
+            deposit_number: nil,
+            reconciled_note: nil,
+            updated_at: Time.current
+          )
```

Comment:
What's the reason to skip callbacks? I think we should use the state machine, adding a new transition if necessary

Also, the only reason I can see this can fail is due to some db error unrelated to the application